src/js/components/Util/TDescriptions.tsx

```diff
@@ -1,13 +1,24 @@
 import { Descriptions } from 'antd';
 import type { DescriptionsProps } from 'antd';
+import type { ConditionalDescriptionItem } from '@/types/descriptions';
 
 import { useTranslatedDescriptionItems } from '@/hooks/useTranslatedDescriptionItems';
+import { hiddenDescriptions } from '@/utils/descriptions';
+
+interface TDescriptionsProps extends Omit<DescriptionsProps, 'items'> {
+  items: ConditionalDescriptionItem[];
+}
 
 // T stands for Translated
-const TDescriptions = (props: DescriptionsProps) => {
+const TDescriptions = (props: TDescriptionsProps) => {
   const { items, ...restProps } = props;
+  const filteredItems = hiddenDescriptions(items);
```

Comment:
this should be memoized based on items I think

Reply:
or hidden descriptions should be a hook itself if all use cases are directly in components like this

Reply:
this component is getting rerendered only when items change, the others are constant display settings. use memo isn't going to do anything. Also the function is only used here and also doesn't really qualify to be a hook, as it internally doesn't use a different hook.